### PR TITLE
Icu 12590 add filtering pagination support for roles

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12590

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12590)

## Description
Add search & pagination to roles.

## Screenshots (if appropriate):
image

## How to Test

* start boundary dev instance
* navigate to roles page
* use search box to find roles using id, name, or description

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- [X] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/admin/app/routes/scopes/scope/roles/index.js [hunk 1]
```diff
@@ -4,9 +4,87 @@
...
+    let totalItems = 0;
+    if (this.can.can('list model', scope, { collection: 'roles' })) {
+      roles = await this.store.query('role', {
+        query: { search, filters },
+        page,
+        pageSize,
+      });
+      totalItems = roles.meta?.totalItems;
+    }
+    await this.getRolesExist(scope_id, totalItems);
```
Comment (ZedLi) [inline ui/admin/app/routes/scopes/scope/roles/index.js:55]: Why do we save this to a class variable `rolesExist` and then return it? Why couldn't you have just done `const rolesExist = this.getRolexExist` and have the result returned from the function?

**Review by ZedLi — APPROVED** [review]
Looks great!

Unrelated to your changes but I noticed that in the individual role if you don't have any permissions for any of the dropdown selections we still show the `Manage` button even if you can't do anything with it
image
I bet this occurs in most of our pages so something to look into later

File: ui/admin/app/routes/scopes/scope/roles/index.js (hunk 1 above)
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/roles/index.js:55]: ah yes good point, ill refactor to do that instead!

Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/roles/index.js:55]: In reference to the manage dropdown: good catch, ur right that happens for roles + other resources. I added this [backlog ticket](https://hashicorp.atlassian.net/browse/ICU-13185) to fix this.

**Review by cameronperera — APPROVED** [review]

**Pull request merged**